src/classes/BDI_Donations.cls

```diff
@@ -143,13 +143,16 @@ public class BDI_Donations {
     
         // get our matching behavior to use
         String matchBehavior = bdi.diSettings.Donation_Matching_Behavior__c;
-
+        System.debug('In Import Donations');
         // get the list of DI records that have donations to process
         List<DataImport__c> dataImportDonations = validateDonationsToProcess(matchBehavior);
 
+        List<DataImport__c> dataImportForMatching = filterOutElevatePaymentDIs(dataImportDonations);
+
         // call the extension to match against existing donations
-        if (matchBehavior != BDI_DataImport_API.DoNotMatch) {
+        if (matchBehavior != BDI_DataImport_API.DoNotMatch && dataImportForMatching.size() > 0) {
             BDI_PerfLogger.PerfLog logger = BDI_DataImportService.perfLogger.newPerfLog('iMatchDonations.match');
+            System.debug('Calling Matching logic for Donations');
             iMatchDonations.match(bdi, dataImportDonations, opportunitiesByDataImportId, paymentsByDataImportId);
```

Comment:
iMatchDonations.match is taking in `dataImportDonations`, is it supposed to take in `dataImportForMatching` instead?

Reply:
@KietTruongSFDO fixed this.